packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -1304,34 +1832,32 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
       );
     }
 
-    double endPadding;
-    switch (textDirection) {
-      case TextDirection.rtl:
-        endPadding = mediaQuery.padding.left;
-        break;
-      case TextDirection.ltr:
-        endPadding = mediaQuery.padding.right;
-        break;
-    }
-    assert(endPadding != null);
-
+    // The minimum insets for contents of the Scaffold to keep visible.
+    final EdgeInsets minInsets = mediaQuery.padding.copyWith(
```

Comment:
@Hixie @HansMuller 

Please take a look at this change.  We discussed it separately, but brief recap:


> What's happening is that the top, end and bottom of mediaQuery.padding are all getting passed to the scaffold layout
But with start-aligned fabs, we really need the start padding as well
and so at this point it's easier to just pass the padding directly and find the start and end padding in the relevant positioner